File: 2024-2/sesiones-de-clase/sesion-de-clase-20-1/AlumnoRegular.cpp
/* 
 * File:   AlumnoRegular.cpp
 * Author: erichuiza
 * 
 * Created on November 11, 2024, 3:18 PM
 */

#include "AlumnoRegular.h"

AlumnoRegular::AlumnoRegular() : Alumno() {
    annoIngreso = {};
}

AlumnoRegular::AlumnoRegular(const AlumnoRegular& orig) : Alumno(orig) {
    *this = orig;
}

Alumno* AlumnoRegular::clonar() {
    return new AlumnoRegular(*this);
}

void AlumnoRegular::operator=(const AlumnoRegular& orig) {
    Alumno::operator =(orig);
    annoIngreso = orig.annoIngreso;
}

void AlumnoRegular::cargar(ifstream& archivo) {
    Alumno::cargar(archivo);
    archivo >> annoIngreso;
}

void AlumnoRegular::imprimir(ofstream& archivo) const {
    Alumno::imprimir(archivo);
    archivo << setw(5) << annoIngreso << endl;
}

AlumnoRegular::~AlumnoRegular() {
}

File: 2025-2/sesion-de-clase-26/src/AlumnoRegular.java

import java.util.Scanner;

/**
 *
 * @author erichuiza
 */
class AlumnoRegular extends Alumno {
    private int annoIngreso;
    
    public AlumnoRegular() {
        super();
    }
    
    public AlumnoRegular(char tipo) {
        super(tipo);
    }
    
    public void setAnnoIngreso(int annoIngreso) {
        this.annoIngreso = annoIngreso;
    }

    public int getAnnoIngreso() {
        return annoIngreso;
    }

    @Override
    public void cargar(Scanner scanner) {
        super.cargar(scanner);
        annoIngreso = scanner.nextInt();
    }

    @Override
    public void imprimir() {
        super.imprimir();
        System.out.println("Año de ingreso: " + annoIngreso);
    }
}
